### PR TITLE
Change `len` from `usize` to `u32`.

### PR DESCRIPTION
**Pull request by pepyakin**
I don't think this is particularly important, but shouldn't harm either.

### CODE REVIEW

**Review by coriolinus — APPROVED** [review]

**Review by drahnr — APPROVED** [review]

**Review by drahnr — APPROVED** [review]

File: parachain/test-parachains/adder/src/wasm_validation.rs [hunk 1]
```diff
@@ -23,8 +23,8 @@ use parachain::primitives::{ValidationResult, HeadData as GenericHeadData};
 use codec::{Encode, Decode};
 
 #[no_mangle]
-pub extern "C" fn validate_block(params: *const u8, len: usize) -> u64 {
-	let params = unsafe { parachain::load_params(params, len) };
+pub extern "C" fn validate_block(params: *const u8, len: u32) -> u64 {
+	let params = unsafe { parachain::load_params(params, len as u32) };
```
Comment (niklasad1) [inline parachain/test-parachains/adder/src/wasm_validation.rs:27]: redundant cast.

```suggestion
	let params = unsafe { parachain::load_params(params, len) };
```

File: parachain/test-parachains/adder/src/wasm_validation.rs [hunk 2]
```diff
@@ -24,7 +24,7 @@ use codec::{Encode, Decode};
 
 #[no_mangle]
 pub extern "C" fn validate_block(params: *const u8, len: u32) -> u64 {
-	let params = unsafe { parachain::load_params(params, len) };
+	let params = unsafe { parachain::load_params(params, len as u32) };
```
Comment (coriolinus) [inline parachain/test-parachains/adder/src/wasm_validation.rs:27]: I don't understand the motivation here if `len` is already `u32` per the function definition.

Comment (pepyakin) [reply parachain/test-parachains/adder/src/wasm_validation.rs:27]: oh, I meant to cast it by `usize` 🙈 , I need to devote more attention to this PR.

**Comment by pepyakin** [comment]
Hold on guys, the third time is the charm

**Review by bkchr — CHANGES_REQUESTED** [review]
If you want to do this correctly, please change the `load_params' function ;)

Otherwise this pr doesn't bring any value.

**Comment by pepyakin** [comment]
Ok, I see your concern, had the same one. I chose the presented solution because the body of `load_params` requires `usize` anyway. So this would just move the cast down. 

Also note, that this is not a destructive cast, since we never assume that the word size is below 32 bits. But if it were, then moving the cast won't help either, we would need to panic to somehow signal this destruction. 

And finally, I **would have** argued that this PR has some value, something along the lines that the risk of UB coming from a broken signature is worse than potential trimming problems in 16 bits platforms yadyadyadya (🙈), but I am not myself completely sure that the root problem is a sufficiently realistic scenario. 

So I am not sure if I am going to persue this PR. Feel free to close it.

**Comment by pepyakin** [comment]
Oh, and look, the CI has passed xD

**Comment by pepyakin** [comment]
Ok, Basti showed me [another example](https://github.com/paritytech/cumulus/blob/3ef4b34a75739a26bddb89cc4b2751d288812f61/runtime/src/validate_block/mod.rs#L64) where `usize` is used and I also found one in substrate and perhaps there are more.

I am closing this, but if anybody feels motivated enough to pick up, feel free to reopen or open another PR

**Pull request closed without merging**